### DISCUSSION TITLE
How to read the string value of an mbbo record

### DISCUSSION THREAD

**Discussion by whs92** [dls-controls/pythonSoftIOC#105] [Q&A] [answered]
I have created an mbbo record:

```
mbbo_record = builder.mbbOut('SELECTION', 'A', 'B', 'C', 'D', initial_value = 0)
```

I would like to access the string value that the pv is set to some time later, but if I do:

```
mbbo_record.get()
```
Then I get the enum integer value of the record. How can I get the string?

**Comment by Araneidae** [comment, marked as answer]
Interesting question.  I'm not sure you can, certainly not with `mbbo_record.get()`, the internal representation of the value is purely as a number.  On the other hand, `caget(mbbo_record.name, datatype=str)` should work, but it is a rather round the houses way of getting at this value.

Really you're not supposed to want the string, or I guess you need to hang onto the string array when creating the record and do the lookup yourself?  Definitely don't think there's a way that avoids using (internal) channel access.

**Reply by whs92** [reply]
>you need to hang onto the string array when creating the record and do the lookup yourself

I ended up doing this,  it does the job. Thanks!
